examples/extensions/remove_entire_sitelink_campaign_extension_setting.py

```diff
@@ -0,0 +1,230 @@
+        customer_id: the client customer ID.
+        campaign_id: the campaign ID.
+
+    Returns:
+        The created MutateOperation for the sitelink campaign extension
+            setting.
+    """
+    extension_type_enum = client.get_type("ExtensionTypeEnum", version="v6")
+    composite_id = ResourceName.format_composite(
+        campaign_id,
+        extension_type_enum.ExtensionType.Name(
+            extension_type_enum.ExtensionType.SITELINK
+        ),
+    )
+    # Construct the campaign extension setting resource name, in format:
+    # customers/{customer_id}/campaignExtensionSettings/{campaign_id}~{extension_type}
+    resource_name = client.get_service(
+        "CampaignExtensionSettingService", version="v6"
+    ).campaign_extension_setting_path(customer_id, composite_id)
+
+    # Create a MutateOperation for the campaign extension setting.
+    mutate_operation = client.get_type("MutateOperation", version="v6")
+    mutate_operation.campaign_extension_setting_operation.remove = resource_name
+    return mutate_operation
+
+
+def _get_all_sitelink_extension_feed_items(
+    client, ga_service, customer_id, campaign_id
+):
+    """Gets all sitelink extension feed items associated to the specified
+    campaign extension setting.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        ga_service: the Google Ads API service client.
+        customer_id: the client customer ID.
+        campaign_id: the ID of the campaign to get the sitelink extension feed
+            items from.
+
+    Returns:
+        An array of str resource names of extension feed items.
+    """
+    campaign_resource_name = client.get_service(
+        "CampaignService", version="v6"
+    ).campaign_path(customer_id, campaign_id)
+    extension_type_enum = client.get_type("ExtensionTypeEnum", version="v6")
+    extension_type_name = extension_type_enum.ExtensionType.Name(
+        extension_type_enum.ExtensionType.SITELINK
+    )
+
+    # Construct the query.
+    query = f"""
+        SELECT
+            campaign_extension_setting.campaign,
+            campaign_extension_setting.extension_type,
+            campaign_extension_setting.extension_feed_items
+        FROM campaign_extension_setting
+        WHERE campaign_extension_setting.campaign = '{campaign_resource_name}'
+        AND campaign_extension_setting.extension_type =
+            '{extension_type_name}'"""
+
+    # Issue a search request using streaming.
+    response = ga_service.search_stream(customer_id, query=query)
+    extension_feed_item_resource_names = []
+    # Iterate through each row and append the extension feed item resource
+    # names to the return array.
+    for batch in response:
+        for row in batch.results:
+            extension_feed_item_resource_names += (
+                row.campaign_extension_setting.extension_feed_items
+            )
+
+    if len(extension_feed_item_resource_names) == 0:
+        raise Exception(
```

Comment:
should specify the type of exception more precisely

Reply:
Do you think `ValueError` makes sense here?

Reply:
yes

Reply:
Does the golden raise an exception? I'm kind of curious why if so, because nothing will catch it. Why not just print a message about why the example can't complete and exit with 0?

Reply:
the golden raises an InvalidArgumentException. The whole runExample method in the golden is within a try [block](https://github.com/googleads/google-ads-php/blob/8a8bd8efbae248c3f80cdadacc7ca10a38b6cd8e/examples/Extensions/RemoveEntireSitelinkCampaignExtensionSetting.php#L73)

Reply:
I think @BenRKarl's suggestion makes sense. So how about in the `if` block:
```
print("The specified campaign does not contain a sitelink campaign extension setting.")
sys.exit(0)
```